### PR TITLE
Add: Negative sign if count is negative

### PR DESCRIPTION
**Pull request by BKSpurgeon**
Solution for #3

### CODE REVIEW

**Comment by brandly** [comment]
this isn't quite the solution we want. on the top level, we don't want this leading blank spot:

Screen Shot 2020-01-01 at 9 52 38 PM

we just want `040` as before, except we'd like to also display a negative sign when needed. in the path of code that builds up the bitmaps for this display, you're currently waiting til the last moment and then tacking on a `-` or a ` `. instead, let's catch the problem earlier.

you'll find some code like this.

```elm
str =
    minLen 3 (String.fromInt n)
```

`String.fromInt` does more than we're giving it credit for. it already gives us what we need.

```elm
String.fromInt -4 == "-4`
```

so what happens to that `-`? well, we feed these characters thru `toInt`, which is defined as follows.

```elm
toInt string =
    case String.toInt string of
        Just num ->
            num

        Nothing ->
            0
```

`String.toInt "-"` will fail to parse, so the `-` is abandoned and a `0` is left in its place. maybe you could intervene more directly in this chain of code that goes from an `Int` to a `List (Element msg)` called `children`.

---

we're currently going from an `Int` like `-4` to a `String` `"-4"`, parsing those substrings like `"4"` back into an `Int` `4`, and then finding the appropriate `Bitmap`. 

instead of `forInt` maybe we have a `forChar : Char -> Element msg`. then, we should be able to generate the `String` we want to show, `map` over its `Char`s, and find an appropriate `Bitmap` for each.

make sense?

p.s. run `npm run format` keep your code consistently formatted. you can most likely integrate `elm-format` [into your text editor](https://github.com/avh4/elm-format#editor-integration), which i really enjoy having.

File: src/Main.elm [hunk 1]
```diff
@@ -484,17 +484,27 @@ viewDigits n =
...
+                Bitmap.positiveSign
+
         children =
-            String.split "" str
-                |> List.map (toInt >> Bitmap.forInt >> digit >> (\c -> c [] []))
+            (String.split "" str
+                |> List.map (toInt >> Bitmap.forInt >> digit >> (\c -> c [] []))) 
+
+        addNumericalSign =
+            digit addPositiveOrNegativeSign |> (\c -> c [] [])
```
Comment (brandly) [inline src/Main.elm]: just wanted to explain something here. 

you copied `(\c -> c [] [])` from above. in the above case, it's a list of things, so we're `map`ping them into this function. but in this case, you just have a single item.

```diff
  addNumericalSign =
-     digit addPositiveOrNegativeSign |> (\c -> c [] [])
+     (digit addPositiveOrNegativeSign) [] []

**Comment by BKSpurgeon** [comment]
Please consider the above patch.

image

As you can see: -2 translates to "0-2".

`Bitmap.forInt` is now translated to `Bitmap.forChar`.

**Comment by brandly** [comment]
forChar looks good!

We’re splitting a string into a list of strings and then toChar is doing some work to get a Char. I think we could use this instead! https://package.elm-lang.org/packages/elm/core/latest/String#toList

**Comment by BKSpurgeon** [comment]
single_digits

This is the output so far: notice how there are only one/two/three digits shown as required. Should this be the desired behaviour, or should it be padded with some zeros?

**Comment by brandly** [comment]
minLen was handling that. Are you still using it?

**Comment by BKSpurgeon** [comment]
Try the lastest one now!

If you're happy I'll squash those commits and overwrite this branch by force.

**Comment by brandly** [comment]
this is really nice! i'm glad we simplified some old code and improved functionality.

let me know when you're ready to merge

**Comment by BKSpurgeon** [comment]
Hi Matt

I have squashed into a single commit. You should see it updated above.

regards

Ben

**Pull request merged**